### ISSUE TITLE
Example for nodeJs fetching OAuth credentials?

### ISSUE THREAD

**Issue by ktwbc** [CrowdStrike/falconjs#191]
Is there a way to get an example of nodeJs calls and showing where it takes the original API key and fetches the secondary OAuth credentials? In attempting to do a simple test, this is failing with an Authorization error but I can confirm through logging that the original ClientId/Secret works and I can manually get an Authtoken yet the subsequent call still fails.

```
    client = falconClientFromEnv(); // same as existing example

    // manual attempt to get token
    let authToken = await client.oauth2.oauth2AccessToken(
      process.env.FALCON_CLIENT_ID as string,
      process.env.FALCON_CLIENT_SECRET as string
    );
    console.log(authToken.accessToken); // does log token so my original ID/Secret works
    const result = await client.falconxSandbox.uploadSampleV2(blob, 'test4.pdf'); // get access denied, invalid bearer token\

```

**Comment by nwesoccer** [comment]
@ktwbc facing the same issue, were you ever able to solve it?

**Comment by ktwbc** [comment]
> @ktwbc facing the same issue, were you ever able to solve it?

No, never did, and abandoned the effort. Crowdstrike said the contributor of this library wasn't a CS employee, had no info on it, and posts to the OP were going unanswered, so I just gave up.

**Comment by nwesoccer** [comment]
@ktwbc ok thanks, it's definitely weird considering it's the client they mention to use for node in their documentation.  https://falcon.crowdstrike.com/documentation/338/falcon-sdks#falcon-js

**Comment by nwesoccer** [comment]
@ktwbc Sorry to spam you, but we were able to find the bug that causes this issue for us, maybe it'll help you too.  The issue is that we are using a "Cloud" that is not the default us-1.  In that scenario this line of code does not set the base path: https://github.com/CrowdStrike/falconjs/blob/main/src/client.ts#L143.  So the initial request to get the token goes to the correct cloud, but the subsequent api requests go to the default us-1 cloud.

The way we are getting around this bug is by not using the "simple" client.  Here is an example:

```
const oauth2 = new OAuth2({
  fetchApi: fetch,
  cloud: "us-2",
  clientId: "",
  clientSecret: "",
});

const config = new Configuration({
  fetchApi: fetch,
  accessToken: oauth2.accessToken.bind(oauth2),
  middleware: [new UserAgent()],
  basePath: CloudBasePath("us-2"),
});

const hostsClient = new HostsApi(config);
```

I'd submit a pull request to fix, but I'm not sure there is anyone maintaining the repo to approve and definitely no one to create a new release.  Hope this helps.

**Comment by TheHinsh** [comment]
After running into a similar issue, I was able to reproduce this error using different curl commands.

```
curl -X GET "https://api.us-2.crowdstrike.com/sensors/entities/datafeed/v2?appId=eng-test2&format=json" -H "accept: application/json" -H "authorization: Bearer validToken"
```

If I change that command from `[api.us-2.crowdstrike.com](https://api.us-2.crowdstrike.com/)` to `[api.crowdstrike.com](https://api.crowdstrike.com/)` and add the `-L` option:
```
curl -X GET "https://api.crowdstrike.com/sensors/entities/datafeed/v2?appId=eng-test2&format=json" -H "accept: application/json" -H "authorization: Bearer validToken" -L -vvv
```

The response back is a 401 - unauthorized.  

Digging into the npm module, it turns out that the `basePath` is not configured using the `cloud` configuration item when making API calls.  It is only used in the `oauth2` configuraiton.

After modifying the `./dist/client.js` with the following... 
```
const FalconCloud_1 = require("./FalconCloud"); // <-- new line
class FalconClient {
    constructor(options) {
        this.options = options;
        const oauth2 = new middleware_1.OAuth2({
            fetchApi: options.fetchApi || fetch,
            cloud: options.cloud,
            basePath: options.basePath || FalconCloud_1.CloudBasePath(options.cloud), // <-- new line
            clientId: options.clientId,
            clientSecret: options.clientSecret,
            memberCid: options.memberCid,
        });
        this.config = new runtime_1.Configuration({
            basePath: options.basePath || FalconCloud_1.CloudBasePath(options.cloud), // <-- new line
            fetchApi: options.fetchApi,
            accessToken: oauth2.accessToken.bind(oauth2),
            middleware: [new middleware_1.UserAgent()],
        });
...
```
I'm able to use the following API call. 
```
  let config = {
      fetchApi: fetch,
      cloud: 'us-2',
      clientId: ClientId,
      clientSecret: Secret,
  };
  let falcon = new FalconClient(config);
  await falcon.availableEventStreams(ClientName).then(streams => console.log(`Got streams... ${streams.length}`);
```

**Comment by MMChrisHinshaw** [comment]
Any news on a new release of this module?

**Comment by nathancharles** [comment]
@jlangdev I just opened a PR that should fix this issue

**Comment by ZacHigi** [comment]
@jlangdev confirming auth issues when using us-2 and oauth

**Comment by evanstoner** [comment]
@ZacHigi 0.2.5 was released last week with the cloud base path fix from #209. Please give it a try and let us know.
